pkg/vbox/vbox_driver.go

```diff
@@ -0,0 +1,230 @@
+	}
+	if *l.Yaml.MountType != limayaml.REVSSHFS {
+		return fmt.Errorf("field `mountType` must be %q for VBox driver , got %q", limayaml.REVSSHFS, *l.Yaml.MountType)
+	}
+	return nil
+}
+
+func (l *LimaVBoxDriver) CreateDisk() error {
+	if err := EnsureDisk(l.BaseDriver); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (l *LimaVBoxDriver) create(ctx context.Context, name string) error {
+	qExe := "VBoxManage"
+
+	qArgsFinal := []string{"createvm", "--basefolder", l.Instance.Dir, "--name", name, "--register"}
+	qCmd := exec.CommandContext(ctx, qExe, qArgsFinal...)
+	_, err := qCmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+	logrus.Debugf("qCmd.Args: %v", qCmd.Args)
+	if err := qCmd.Run(); err != nil {
+		return err
+	}
+
+	baseDisk := filepath.Join(l.Instance.Dir, filenames.BaseDisk)
+	diffDisk := filepath.Join(l.Instance.Dir, filenames.DiffDisk)
+	isBaseDiskISO, err := iso9660util.IsISO9660(baseDisk)
+	if err != nil {
+		return err
+	}
+
+	var firmware string
+	if *l.Yaml.Firmware.LegacyBIOS {
+		firmware = "bios"
+	} else {
+		firmware = "efi"
+	}
+	cpus := *l.Yaml.CPUs
+	memBytes, err := units.RAMInBytes(*l.Yaml.Memory)
+	if err != nil {
+		return err
+	}
+	memory := memBytes >> 20
+	var boot string
+	if isBaseDiskISO {
+		boot = "dvd"
+	} else {
+		boot = "disk"
+	}
+
+	var modifyFlags = []string{
+		"modifyvm", name,
+		"--firmware", firmware,
+		"--ostype", "Linux26_64",
+		"--cpus", fmt.Sprintf("%d", cpus),
+		"--memory", fmt.Sprintf("%d", memory),
+		"--boot1", boot,
+	}
+	mCmd := exec.CommandContext(ctx, qExe, modifyFlags...)
+	logrus.Debugf("mCmd.Args: %v", mCmd.Args)
+	if err := mCmd.Run(); err != nil {
+		return err
+	}
+
+	logrus.Debugf("storage")
+	if err := exec.CommandContext(ctx, qExe, "storagectl", name,
+		"--name", "SATA",
+		"--add", "sata",
+		"--portcount", "4",
+		"--hostiocache", "on").Run(); err != nil {
+		logrus.Debugf("storagectl %v", err)
+		return err
+	}
+	if isBaseDiskISO {
+		if err := exec.CommandContext(ctx, qExe, "storageattach", name,
+			"--storagectl", "SATA",
+			"--port", "1",
+			"--device", "0",
+			"--type", "dvddrive",
+			"--medium", baseDisk+".iso").Run(); err != nil {
+			logrus.Debugf("basedisk %v", err)
+			return err
+		}
+	}
+	if err := exec.CommandContext(ctx, qExe, "storageattach", name,
+		"--storagectl", "SATA",
+		"--port", "0",
+		"--device", "0",
+		"--type", "hdd",
+		"--medium", diffDisk+".vdi").Run(); err != nil {
+		logrus.Debugf("diffdisk %v", err)
+		return err
+	}
+
+	if err := exec.CommandContext(ctx, qExe, "storageattach", name,
+		"--storagectl", "SATA",
+		"--type", "dvddrive",
+		"--port", "2",
+		"--device", "0",
+		"--medium", filepath.Join(l.Instance.Dir, filenames.CIDataISO)).Run(); err != nil {
+		logrus.Debugf("cidata %v", err)
+		return err
+	}
+
+	logrus.Debugf("network")
+
+	slirpMACAddress := limayaml.MACAddress(l.Instance.Dir)
+	if out, err := exec.CommandContext(ctx, qExe, "modifyvm", name,
+		"--nic1", "nat",
```

Comment:
Just putting it out.

Is vpn connection from vm is working??
Also please check if our hostResolver works as well.

In vz, with nat we had issues with these, i believe same should happen here as well. 
This was the main reason why we used gvisor-tap-vsock to setup slirp network

Reply:
It does not, there are separate flags for host resolver afaik.

Early days yet, this was more of a hack to see if it worked...